vllm/core/scheduler.py

```diff
@@ -101,6 +102,118 @@ def num_curr_seqs(self):
         return self._num_curr_seqs
 
 
+@dataclass
+class PaddingAwareSchedulingBudget(SchedulingBudget):
+    max_num_prefill_seqs: Optional[int] = None
+    _prefill_request_ids_max_seq_lens: Dict[str,
+                                            int] = field(default_factory=dict)
+    _max_seq_len: int = 0
+    _num_curr_prefill_seqs: int = 0
+
+    def _debug_validator(self):
+        assert self._num_curr_prefill_seqs == len(
+            self._prefill_request_ids_max_seq_lens)
+        if self._num_curr_prefill_seqs != 0:
+            assert self._max_seq_len == max(
+                self._prefill_request_ids_max_seq_lens.values())
+
+    def _generic_padding_fn(self, batch_size, max_seq_len) -> int:
+        return batch_size * max_seq_len
+
+    def _hpu_padding_fn(self, batch_size, max_seq_len):
+        from vllm.worker.hpu_model_runner import (HPUBucketingGlobalState,
```

Comment:
this function is not called once yes? is there a reason to put import inside method here and not on top? I guess we dont get any performance penalty here since you are not doing a lookup each time, just use the imported class/method, but am not 100% sure